src/dom-to-image-more.js

```diff
@@ -257,24 +258,38 @@
         return Promise.resolve(node)
             .then(makeNodeCopy)
             .then(function (clone) {
-                return cloneChildren(node, clone);
+                return cloneChildren(util.hasOpenShadowRoot(node) ? node.shadowRoot : node, clone);
             })
             .then(function (clone) {
                 return processClone(node, clone);
             });
 
         function makeNodeCopy(original) {
-            return util.isHTMLCanvasElement(original)
-                ? util.makeImage(original.toDataURL())
-                : original.cloneNode(false);
+            if (util.isHTMLCanvasElement(original)) {
+                return util.makeImage(original.toDataURL());
+            }
+            if (util.isInShadowRoot(original)) {
+                return cloneShadowNode(original);
+            }
+            return original.cloneNode(false);
+        }
+
+        function cloneShadowNode(original) {
+            const nextSibling = original.nextSibling;
+            const parentNode = original.parentNode;
+            // Bypass element encapsulation, by detaching from shadow node then restoring.
+            parentNode.removeChild(original);
+            const clone = original.cloneNode(false);
+            parentNode.insertBefore(original, nextSibling);
+            return clone;
         }
 
         function cloneChildren(original, clone) {
-            const originalChildren = original.childNodes;
+            const originalChildren = util.isShadowSlotElement(original) ? original.assignedNodes() : original.childNodes;
```

Comment:
Is `original.assignedNodes()` null or just zero-lengthed if this is NOT a shadow slot element?

Reply:
It's null :s

Reply:
It's null :s

Reply:
Abstracted away and added a comment in [d656297](https://github.com/1904labs/dom-to-image-more/pull/101/commits/d65629752cbacad624ac790d3bec9980b553361c)